pkgs/applications/networking/feedreaders/castget/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchFromGitHub, autoreconfHook, pkgconfig
+, glib, ronn, curl, id3lib, libxml2 }:
+
+stdenv.mkDerivation rec {
+  pname = "castget";
+  version = "1_2_4";
+
+  src = fetchFromGitHub {
+    owner = "mlj";
+    repo = pname;
+    rev = "rel_${version}";
+    sha256 = "1pfrjmsikv35cc0praxgim26zq4r7dfp1pkn6n9fz3fm73gxylyv";
+  };
+  prePatch = ''
+    touch NEWS
```

Comment:
Why is this necessary?

Reply:
Otherwise, `autoreconf` fails with:

```
Makefile.am: error: required file './NEWS' not found
Makefile.am: error: required file './README' not found
Makefile.am: error: required file './ChangeLog' not found
```

Reply:
Weird that they don't mention that in the README. Can you replace the `touch` by `automake --add-missing` to be more robust and better match the intent? Also, this should be done in `preAutoreconf` instead of `prePatch` since its an extension to the autoreconf phase.

Reply:
Did you mean to run `automake --add-missing` in the `preAutoreconf` hook? That fails with the same errors. It seems from the log that this command is run with these flags by the autoreconf hook itself:

```
autoreconf: running: /nix/store/0yl3mx4f8f1bnmyfzxv6a3bik7c2ni87-autoconf-2.69/bin/autoconf --force
autoreconf: running: /nix/store/0yl3mx4f8f1bnmyfzxv6a3bik7c2ni87-autoconf-2.69/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
```

Using the same `touch` commands but only using `preAutoreconf` instead of `prePatch` works as well.

From AFAIK, the requirement for these files is a GNU strictness that probably stayed there from the old days this software was distributed by them. The release is from 2017 so I'm not surprised it's not documented.

See this: https://www.gnu.org/software/automake/manual/automake.html#index-gnu-strictness

Reply:
That's weird, especially since castget is marked as `foreign`. Anyways, its not that important.